web/src/ProductSelector.js

```diff
@@ -11,38 +11,63 @@ import {
   ModalVariant
 } from "@patternfly/react-core"
 
-export default function ProductSelector({ value, onChange = () => {} }) {
-  const [isFormOpen, setFormOpen] = useState(false);
-  const [product, setProduct] = useState(value);
-  const [products, setProducts] = useState([]);
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "LOAD": {
+      return { ...state, ...action.payload };
+    }
+    case "ACCEPT": {
+      return { ...state, isFormOpen: false };
+    }
+
+    case "CANCEL": {
+      return { ...state, isFormOpen: false, current: state.initial };
+    }
+
+    case "CHANGE": {
+      return { ...state, current: action.payload };
+    }
+
+    case "OPEN": {
+      return { ...state, isFormOpen: true };
+    }
+
+    default: {
+        return state;
+    }
+  }
+}
+
+const initialState = {
+  products: [], initial: null, current: null, isFormOpen: false
+};
+
+export default function ProductSelector() {
```

Comment:
Note for myself: looking at the similarities of current selectors, looks like could be a good idea to build a `FormSelector` component. But maybe it is too soon yet.